FactorioClient.py

```diff
@@ -123,6 +132,38 @@ def on_package(self, cmd: str, args: dict):
                                      f"{Utils.format_SI_prefix(args['value'])}J remaining.")
                         self.rcon_client.send_command(f"/ap-energylink {gained}")
 
+    def on_user_say(self, text: str) -> typing.Optional[str]:
+        # Mirror chat sent from the UI to the Factorio server.
+        self.print_to_game(f"{self.player_names[self.slot]}: {text}")
+        return text
+
+    async def chat_from_factorio(self, user: str, message: str) -> None:
+        if not self.bridge_chat_out:
+            return
+
+        # Pass through commands
+        if message.startswith("!"):
+            await self.send_msgs([{"cmd": "Say", "text": message}])
+            return
+
+        # Remove local coordinates
+        message = re.sub(r"\[gps=[^\]]*\]", "", message).strip()
```

Comment:
if we assume all messages with `[gps=` are pings and should not get sent, we can do an easier, faster and more readable
`if "[gps=" in message: return`

Reply:
If you wish to assume that, we can do that, sure. Note that it is also possible to have a mixed chat message by pinging with an open chat box. One might write "i have a lot of iron over here [gps=4711,815]".

Reply:
yeah, my thinking is that a ping should never concern players outside the factorio session

Reply:
My thinking was that removing the entire mixed message could lead to confusion when replies to such messages are bridged.

Reply:
i think `see iron here [gps=]` is the most common scenario, which is worse than having replies you don't understand